### PR TITLE
Fix 0.20.3

### LINKED ISSUE #205: polars 0.20.3 module has no members
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

It appears 0.20.3 builds a unusable module with no members, at least on linux x64:

```
Python 3.12.1 | packaged by conda-forge | (main, Dec 23 2023, 08:03:24) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import polars
>>> polars.DataFrame
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'polars' has no attribute 'DataFrame'
>>> dir(polars)
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
>>> 
```

0.20.2 doesn't have this issue. The pip wheel for 0.20.3 doesn't have this issue, either.

### Installed packages

```shell
# packages in environment at /home/lidavidm/miniforge3/envs/polars:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2023.11.17           hbcca054_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libblas                   3.9.0           20_linux64_openblas    conda-forge
libcblas                  3.9.0           20_linux64_openblas    conda-forge
libexpat                  2.5.0                hcb278e6_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_3    conda-forge
libgfortran-ng            13.2.0               h69a702a_3    conda-forge
libgfortran5              13.2.0               ha4646dd_3    conda-forge
libgomp                   13.2.0               h807b86a_3    conda-forge
liblapack                 3.9.0           20_linux64_openblas    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.25          pthreads_h413a1c8_0    conda-forge
libsqlite                 3.44.2               h2797004_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_3    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
ncurses                   6.4                  h59595ed_2    conda-forge
numpy                     1.26.3          py312heda63a1_0    conda-forge
openssl                   3.2.0                hd590300_1    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pip                       23.3.2             pyhd8ed1ab_0    conda-forge
polars                    0.20.3          py312hfa2e56e_0    conda-forge
python                    3.12.1          hab00c5b_1_cpython    conda-forge
python_abi                3.12                    4_cp312    conda-forge
readline                  8.2                  h8228510_1    conda-forge
setuptools                69.0.3             pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tzdata                    2023d                h0c530f3_0    conda-forge
wheel                     0.42.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
```

### Environment info

```shell
active environment : polars
    active env location : /home/lidavidm/miniforge3/envs/polars
            shell level : 2
       user config file : /home/lidavidm/.condarc
 populated config files : /home/lidavidm/miniforge3/.condarc
                          /home/lidavidm/.condarc
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.10.12.final.0
       virtual packages : __archspec=1=x86_64
                          __cuda=12.2=0
                          __glibc=2.36=0
                          __linux=6.1.0=0
                          __unix=0=0
       base environment : /home/lidavidm/miniforge3  (writable)
      conda av data dir : /home/lidavidm/miniforge3/etc/conda
  conda av metadata url : None
                          https://conda.anaconda.org/rapidsai/linux-64
                          https://conda.anaconda.org/rapidsai/noarch
                          https://conda.anaconda.org/nvidia/linux-64
                          https://conda.anaconda.org/nvidia/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/lidavidm/miniforge3/pkgs
                          /home/lidavidm/.conda/pkgs
       envs directories : /home/lidavidm/miniforge3/envs
                          /home/lidavidm/.conda/envs
               platform : linux-64
             user-agent : conda/23.3.1 requests/2.31.0 CPython/3.10.12 Linux/6.1.0-12-amd64 debian/12 glibc/2.36
                UID:GID : 1000:1000
             netrc file : /home/lidavidm/.netrc
           offline mode : False
```

### PR DESCRIPTION
**Pull request by pavelzw**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #205

### CODE REVIEW

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

**Comment by 0xbe7a** [comment]
Linux-64 generated valid conda package locally with `conda-build 3.28.2`

**Comment by pavelzw** [comment]
Let's also run pytest in the test section as suggested in https://github.com/conda-forge/polars-feedstock/issues/205#issuecomment-1876786385
Other larger feedstocks such as scipy, scikit-learn also do this.

Then we can at least be somewhat sure about the Linux and osx-64 builds. Win-64 is unfortunately not testable due to cross-compilation.

**Comment by 0xbe7a** [comment]
The wheels created by `maturin build --strip` are already broken and contain only the .so binary

File: .gitignore
Comment (pavelzw) [inline .gitignore:1]: This needs to be changed manually on each rerender until https://github.com/conda-forge/conda-smithy/issues/1827 is fixed.

File: conda-forge.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ azure:
   free_disk_space: true
   settings_linux:
     swapfile_size: 10GiB
+  store_build_artifacts: True
```
Comment (0xbe7a) [inline conda-forge.yml:5]:
```suggestion
```

**Review by 0xbe7a — APPROVED** [review]

**Pull request merged**